server/server.go

```diff
@@ -0,0 +1,58 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"PhilomathesInc/demo-ci-pipelines/config"
+
+	"go.uber.org/zap"
+)
+
+const (
+	healthPath  = "/health"
+	welcomePath = "/"
+)
+
+func Run(c *config.Config) {
+	c.Logger.Info("initializing the server...")
+	http.HandleFunc(healthPath, healthHandler(c))
+	http.HandleFunc(welcomePath, welcomeHandler(c))
+
+	timeout, err := time.ParseDuration(c.Server.Timeout)
+	if err != nil {
+		c.Logger.Error("error parsing timeout value", zap.String("error", err.Error()))
+	}
+	s := http.Server{
+		Addr:         fmt.Sprintf(":%s", c.Server.Port),
+		ReadTimeout:  timeout,
+		WriteTimeout: timeout,
+	}
+
+	c.Logger.Info(fmt.Sprintf("you can now visit the server on %s:%s", c.Server.Hostname, c.Server.Port))
+	if err := s.ListenAndServe(); err != nil {
+		c.Logger.Fatal("error occured on the HTTP server", zap.String("error", err.Error()))
+	}
+}
+
+func healthHandler(c *config.Config) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, req *http.Request) {
+		c.Logger.Info("received request to check the server health")
+		w.Header().Add("Content-Type", "application/json")
+		response, err := json.Marshal(Healthy)
```

Comment:
There is no need to Marshal the string here. As you're setting the `Content-Type` header to `application/json` already, the response being expected is simply JSON data and the contents of your string `Healthy ` is actually JSON data. You just need to convert it to a byte slice.

```
# Your server response, when you're sending the marshalled data
$ curl -s localhost:9091/health | jq .
"{\"status\":\"ok\"}"
$ curl -s localhost:9091/health | jq .status
jq: error (at <stdin>:0): Cannot index string with string "status"

# As a demonstration for the correct JSON response, see GitHub API
$ curl -s https://api.github.com/users/mugdha-adhav | jq .
{
  "login": "mugdha-adhav",
  "id": 37295843,
  "node_id": "MDQ6VXNlcjM3Mjk1ODQz",
  "avatar_url": "https://avatars.githubusercontent.com/u/37295843?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/mugdha-adhav",
  "html_url": "https://github.com/mugdha-adhav",
  "followers_url": "https://api.github.com/users/mugdha-adhav/followers",
  "following_url": "https://api.github.com/users/mugdha-adhav/following{/other_user}",
  "gists_url": "https://api.github.com/users/mugdha-adhav/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/mugdha-adhav/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/mugdha-adhav/subscriptions",
  "organizations_url": "https://api.github.com/users/mugdha-adhav/orgs",
  "repos_url": "https://api.github.com/users/mugdha-adhav/repos",
  "events_url": "https://api.github.com/users/mugdha-adhav/events{/privacy}",
  "received_events_url": "https://api.github.com/users/mugdha-adhav/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Mugdha Adhav",
  "company": "@acquia ",
  "blog": "https://mugdha.dev",
  "location": "Remote",
  "email": null,
  "hireable": true,
  "bio": "Programmer & Gamer",
  "twitter_username": "AdhavMugdha",
  "public_repos": 12,
  "public_gists": 1,
  "followers": 8,
  "following": 26,
  "created_at": "2018-03-12T11:48:32Z",
  "updated_at": "2021-07-08T05:47:17Z"
}
$ curl -s https://api.github.com/users/mugdha-adhav | jq .login
"mugdha-adhav"

# If you don't marshall and simply send the byte slice of the string, it behaves similar to GitHub API
$ curl -s localhost:9091/health | jq .
{
  "status": "ok"
}
$ curl -s localhost:9091/health | jq .status
"ok"
```

Reply:
Great catch @mbtamuli
Fixing it.
And thanks for the elaborate description, it helped understand and test things quickly.